Src/FluentAssertions/StringContainConstraint.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Diagnostics;
+using FluentAssertions.Primitives;
+
+namespace FluentAssertions
+{
+    [DebuggerNonUserCode]
+    public class StringContainConstraint : AndConstraint<StringAssertions>
+    {
+        public StringContainConstraint(StringAssertions parentConstraint, string expected) : base(parentConstraint)
+        {
+            ParentConstraint = parentConstraint;
+            Expected = expected;
+            Occurrences = CountOccurrences(parentConstraint.Subject, expected, StringComparison.Ordinal);
+        }
+
+        private string Expected { get; }
+        private int Occurrences { get; }
+        private StringAssertions ParentConstraint { get; }
+
+        public StringWithOccurenceConstraint Exactly => new StringWithOccurenceConstraint(ParentConstraint,
+            Expected,
+            Occurrences,
+            "exactly",
+            (occuredTimes, expectedTimes) => occuredTimes == expectedTimes);
+
+        public StringWithOccurenceConstraint AtLeast => new StringWithOccurenceConstraint(ParentConstraint,
+            Expected,
+            Occurrences,
+            "at least",
+            (occuredTimes, expectedTimes) => occuredTimes >= expectedTimes);
+
+        public StringWithOccurenceConstraint MoreThan => new StringWithOccurenceConstraint(ParentConstraint,
+            Expected,
+            Occurrences,
+            "more than",
+            (occuredTimes, expectedTimes) => occuredTimes > expectedTimes);
+
+        public StringWithOccurenceConstraint AtMost => new StringWithOccurenceConstraint(ParentConstraint,
+            Expected,
+            Occurrences,
+            "at most",
+            (occuredTimes, expectedTimes) => occuredTimes <= expectedTimes);
+
+        public StringWithOccurenceConstraint LessThan => new StringWithOccurenceConstraint(ParentConstraint,
+            Expected,
+            Occurrences,
+            "less than",
+            (occuredTimes, expectedTimes) => occuredTimes < expectedTimes);
+
+        public AndConstraint<StringAssertions> Once(string because = "", params object[] becauseArgs) =>
+            Exactly.Once(because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Twice(string because = "", params object[] becauseArgs) =>
+            Exactly.Twice(because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Thrice(string because = "", params object[] becauseArgs) =>
+            Exactly.Thrice(because, becauseArgs);
+
+        public AndConstraint<StringAssertions> Times(int times, string because = "", params object[] becauseArgs) =>
+            Exactly.Times(times, because, becauseArgs);
+
+        private static int CountOccurrences(string actual, string expected, StringComparison comparison)
```

Comment:
🔧 We normally order the members of a method in order of execution so you can read the file as a book (from Clean Code).

Reply:
Ok, my mistake, I used resharper to order the methods.
So please confirm the order:

1. CountOccurrences
2. Once
3. Twice
4. Thrice
5. Times
6. Exactly
7. AtLeast
8. MoreThan
9. AtMost
10. LessThan

Reply:
Correct.